src/controllers/challengesController.ts

```diff
@@ -4,68 +4,110 @@ import { HttpResponseCode } from "../util/errorHandling/httpResponseCode";
 import { getConnection } from "typeorm";
 import { Challenge } from "../db/entity/hub";
 import { Cache } from "../util/cache";
+import { ValidationError, validate } from "class-validator";
+
+// TODO: move into the controller when JS functions are replaced with arrow functions
+let cache: Cache;
 
 export class ChallengesController {
 
-  public async listChallenges(req: Request, res: Response) {
-    res.send(await Cache.challenges.getElements());
+  constructor(_cache: Cache) {
+    cache = _cache;
+  }
+
+  public async listChallenges(req: Request, res: Response, next: NextFunction) {
+    try {
+      let challenges: Challenge[] = cache.getAll(Challenge.name);
+
+      if (challenges.length === 0) {
+        challenges = await getConnection("hub").getRepository(Challenge).find();
+        cache.setAll(Challenge.name, challenges);
+      }
+
+      res.send(challenges);
+    } catch (err) {
+      return next(err);
+    }
   }
 
   public async createChallenge(req: Request, res: Response, next: NextFunction) {
```

Comment:
So since these instance functions they don't have the same problem with losing the context of this and I think they look cleaner than arrow functions. But I do have to admit that they do bring in some inconsistency to the codebase if we're using arrow functions in other places. So I suppose I can switch them over to arrow functions if you want to?

Reply:
From what I have seem in #109, when the function sin the controller are using the services and are not arrow functions, they still lose the `this` context, not exactly sure of the behaviour with it

Reply:
Okay, I was incorrect. Simple JS functions aren't context binding even if they're defined within a class. I'll change them to arrow functions